File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03283/p03283_num6_parsed.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int N, M, Q;
    cin >> N >> M >> Q;

    // create a 2D array to store the train information
    vector<vector<int>> trains(M, vector<int>(2));
    for (int i = 0; i < M; i++) {
        cin >> trains[i][0] >> trains[i][1];
    }

    // process the queries
    for (int i = 0; i < Q; i++) {
        int p, q;
        cin >> p >> q;

        // count the number of trains that run strictly within the section from City p to City q
        int count = 0;
        for (int j = 0; j < M; j++) {
            if (trains[j][0] >= p && trains[j][1] <= q) {
                count++;
            }
        }

        // print the result
        cout << count << endl;
    }

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03283/p03283_num6_parsed.py
def calculate_number_of_trains(N, M, Q, train_routes, queries):
    # Initialize a list to store the number of trains within each section
    train_count = []

    # Iterate over each query
    for query in queries:
        p, q = query

        # Initialize a counter for the number of trains
        count = 0

        # Iterate over each train route
        for route in train_routes:
            L, R = route

            # Check if the train route is within the given section
            if p <= L and R <= q:
                count += 1

        train_count.append(count)

    return train_count


# Read inputs
N, M, Q = map(int, input().split())

train_routes = []
for _ in range(M):
    L, R = map(int, input().split())
    train_routes.append((L, R))

queries = []
for _ in range(Q):
    p, q = map(int, input().split())
    queries.append((p, q))

# Calculate the number of trains within each section
result = calculate_number_of_trains(N, M, Q, train_routes, queries)

# Print the result
for count in result:
    print(count)